processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -470,20 +480,71 @@ public String apply(@Nullable String input)
       }
     };
 
-    return makeIndexFiles(indexes, outDir, progress, mergedDimensions, mergedMetrics, null, rowMergerFn, indexSpec);
+    return makeIndexFiles(
+        indexes,
+        mergeAggregators(indexes),
+        outDir,
+        progress,
+        mergedDimensions,
+        mergedMetrics,
+        null,
+        rowMergerFn,
+        indexSpec
+    );
+  }
+
+  public static final AggregatorFactory[] mergeAggregators(List<IndexableAdapter> indexes)
+  {
+    Set<AggregatorFactory> mergedAggregators = Sets.newTreeSet(
+        Ordering.from(
+            new Comparator<AggregatorFactory>()
+            {
+              @Override
+              public int compare(AggregatorFactory a1, AggregatorFactory a2)
+              {
+                String name1 = a1.getName();
+                String name2 = a2.getName();
+                if (name1 == null) {
+                  return -1;
+                } else {
+                  return name1.compareTo(name2);
+                }
+              }
+            }
+        ).nullsFirst()
+    );
+    for (IndexableAdapter index : indexes) {
```

Comment:
@xvrl @gianm

If 2 segments have aggregators with same name (and type) then it is difficult to decide which one of them to use. I am going to update AggregatorFactory with a new method, something like
```
AggregatorFactory mergingAggregator(AggregatorFactory other) throws NotMergeableException
```
and use that here while merging